Dockerfile

```diff
@@ -0,0 +1,63 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_VERSION=1.23.7
+ARG LLAMA_SERVER_VERSION=latest
+ARG LLAMA_BINARY_PATH=/com.docker.llama-server.native.linux.cpu.amd64
+
+FROM golang:${GO_VERSION}-alpine AS builder
+
+# Install git for go mod download if needed
+RUN apk add --no-cache git
+
+WORKDIR /app
+
+# Copy go mod/sum first for better caching
+COPY --link go.mod go.sum ./
+
+# Download dependencies (with cache mounts)
+RUN --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    go mod download
+
+# Copy the rest of the source code
+COPY --link . .
+
+# Build the Go binary (static build)
+RUN --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    CGO_ENABLED=0 GOOS=linux go build -ldflags="-s -w" -o model-runner ./main.go
+
+# --- Get llama.cpp binary ---
+FROM docker/docker-model-backend-llamacpp:${LLAMA_SERVER_VERSION} AS llama-server
+
+# --- Final image ---
+FROM debian:bookworm-slim AS final
+
+# Create non-root user
+RUN groupadd --system modelrunner && useradd --system --gid modelrunner modelrunner
+
+# Install ca-certificates for HTTPS
+RUN apt-get update && apt-get install -y --no-install-recommends ca-certificates && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /app
+
+# Create directories for the socket file and llama.cpp binary, and set proper permissions
+RUN mkdir -p /var/run/model-runner /app/bin /home/modelrunner/.docker/models && \
```

Comment:
It might be more idiomatic to store model data in `/var/lib/model-runner/models`, or something top-level like `/models`.